packages/checkout/components/text-input/validated-text-input.tsx

```diff
@@ -164,9 +167,21 @@ const ValidatedTextInput = ( {
 		if ( focusOnMount ) {
 			inputRef.current?.focus();
 		}
+
+		// Skip validation only if validateOnMount is false and focusOnMount is true.
+		if ( ! validateOnMount && focusOnMount ) {
```

Comment:
While this might not be the cleanest code, this `if` exists for readability.

We don't want to run `validateInput` if `validateOnMount` is `false` AND `focusOnMount` is true, so we can escape the function early here. Trying to invert this condition to conditionally call `validateInput` results in:

```js
if ( ! ( validateOnMount || !focusOnMount ) ) {
  validateInput( true );
  return;
}
```

and is less-readable in my opinion.

Reply:
I think we're both uneasy with something here, right? Sure, it works, but it got both our spider senses tingling, hence your comment.

I don't know if the snippet you suggested is equivalent, since, in none of the cases it would call only `validateInput()` and exit.
Respectively for `validateOnMount` and `focusOnMount`, you would get the following executions:

- `true`, `true` => validateInput(), setIsPristine()
- `true`, `false` => validateInput(), setIsPristine()
- `false`, `false` => validateInput(), setIsPristine()
- `false`, `true` => setIsPristine()

Isn't that equivalent to:

````
if (validateOnMount || !focusOnMount ) {
    validateInput(true);
}

setIsPristine(false);
````

It should execute things in the same order and amount given the same conditions, right?
I'm sorry if I'm not seeing things clearly and my suggestion is dumb. I did some overthinking and got bamboozled by the original code 😅.  I also like to use the word _bamboozled_ a lot 🙄

It would also be useful to have a unit test for this method 🤔

Reply:
Thanks Paulo, I wrote unit tests with my implementation, ensured they passed, then changed it to yours and they still pass.

The test cases I wrote are:

- `validates when focusOnMount is true and validateOnMount is not set`
- `validates when focusOnMount is false, regardless of validateOnMount value`
- `does not validate when validateOnMount is false and focusOnMount is true`

Please review these new changes when you have chance!

Reply:
I also fixed the `Shows a custom error message for an invalid required input` test which was not making a good assertion, it now checks that the error message is visible on the screen after blurring the input.

Reply:
@wavvves this is ready for review!